.github/workflows/ci.yml

```diff
@@ -98,33 +96,15 @@ jobs:
       - uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
           components: clippy
-          toolchain: 1.70.0
+          toolchain: 1.59.0
```

Comment:
1.7 years old. so can use only rust 1.59 in code?

Reply:
https://github.com/elastio/bon/blob/master/rust-toolchain contains 1.81 . so only CI can check if code is right, not local cargo check?

Reply:
Yep, we need to support older compiler versions to let other libraries that support them use `bon`

Reply:
oh, `step echo '1.59.0' > rust-toolchain`

Reply:
seems very sloppy?

Reply:
It is, but I think it's good enough

Reply:
> [master/rust-toolchain](https://github.com/elastio/bon/blob/master/rust-toolchain?rgh-link-date=2024-09-10T00%3A18%3A29Z) contains 1.81 . so only CI can check if code is right, not local cargo check?

Yes, because the main focus should be on the current version of rust toolchain. For example, `#[diagnostic::on_unimplemented]` isn't available in rust `1.59.0`, so that older version of Rust will get worse compile error messages. But we don't need to work in that environment, we just need some check on CI that our code is compatible with that older version.

Besides there is a [`clippy::incompatible_msrv`](https://rust-lang.github.io/rust-clippy/master/index.html#/incompatible_msrv) lint that validates that the code doesn't use APIs not stable in 1.59.0

Reply:
If you want to check for MSRV locally, you can just manually run `./scripts/test-msrv.sh`. It will run the test in a temp directory